app/sliders/importer.py

```diff
@@ -166,6 +174,48 @@ def init_nested_dict(self):
         :return: defaultdict, a nested dictionary
         """
         return defaultdict(self.init_nested_dict)
+    
+
+class FUEDataParser:
+    def parse(self, sheets):
+        data = self.extract(sheets)
+
+        return data
+    
+    def extract(self, sheets):
+        """
+        Extract Fue Expenses data.
+
+        :param sheets: dict, where keys are names of sheets and values are pandas DataFrames
+        :return: nested dict, with the following shape {area: {year: {x: unit[], y: value[]}}}
+        """
+        extracted = self.init_nested_dict()
+        relevant_years = [2011, 2020]
+        for sheet_key in sheets:
+            basis, _, area = sheet_key.split("_")
+            if basis == "fue":
+                df = sheets[sheet_key]
+                row = df.loc[df["Jahr"].isin(relevant_years)]
+
+                for year in relevant_years:
+                    extracted[area][str(year)] = {"x": [], "y": []}
+                    for unit in mapping_fue:
+                        value = row.iloc[0][unit]
+                        # I used the original key from the spredsheet here, so it get's displayed properly in the Graph.
+                        extracted[area][str(year)]["x"].append(unit)
+                        extracted[area][str(year)]["y"].append(value)
+        return extracted
+    
+    def init_nested_dict(self):
```

Comment:
This function is duplicated. We can take it out of the class so it can be reused by both the basis and the fue ausgaben classes.

Reply:
very good